### PR TITLE
Standardize provider

### PR DESCRIPTION
**Pull request by CiprianDraghici**

### CODE REVIEW

**Review by razvantomegea — DISMISSED** [review]

**Review by arhtudormorar — DISMISSED** [review]

File: src/extensionProvider.ts [hunk 1]
```diff
@@ -63,7 +61,7 @@ export class ExtensionProvider {
     const { token } = options;
     const data = token ? token : "";
     await this.startBgrMsgChannel(Operation.Connect, data);
-    return this.account.address;
+    return this.account;
```
Comment (arhtudormorar) [inline src/extensionProvider.ts:64]: what about signature?

Comment (CiprianDraghici) [reply src/extensionProvider.ts:65]: It's already handled. Please check line:201
`this.account = event.data.data;`

File: src/extensionProvider.ts [hunk 2]
```diff
@@ -1,26 +1,25 @@
-import { IPlainTransactionObject } from "@multiversx/sdk-core/out";
-import { SignableMessage } from "@multiversx/sdk-core/out/signableMessage";
+import { IPlainTransactionObject, Message } from "@multiversx/sdk-core";
```
Comment (radumojic) [inline src/extensionProvider.ts:1]: `Address` can also be imported here instead on another separate import on line 8

File: src/extensionProvider.ts [hunk 3]
```diff
@@ -161,8 +166,13 @@ export class ExtensionProvider {
     const signatureHex = extensionResponse.signature;
     const signature = Buffer.from(signatureHex, "hex");
 
-    message.applySignature(signature);
-    return message;
+    return new Message({
+      data: Buffer.from(messageToSign.data),
+      address: messageToSign.address ?? Address.fromBech32(this.account.address),
+      signer: "extension",
```
Comment (andreibancioiu) [inline src/extensionProvider.ts:172]: Is there a more precise name? E.g. `MultiversX DeFi Wallet`?

Comment (andreibancioiu) [reply src/extensionProvider.ts:172]: Or maybe this provider is used for other extensions, as well.

Comment (CiprianDraghici) [reply src/extensionProvider.ts:172]: Could be used by anybody

File: package.json [hunk 4]
```diff
@@ -15,11 +15,11 @@
...
   "devDependencies": {
+    "@multiversx/sdk-core": ">= 13.5.0",
     "@types/node": "17.0.23",
-    "esmify": "2.1.1",
-    "typescript": "4.1.2"
+    "typescript": "5.5.4"
   },
   "peerDependencies": {
-    "@multiversx/sdk-core": ">= 12.1.0"
+    "@multiversx/sdk-core": ">= 13.5.0"
```
[the remark this answers is not in the document]
Comment (CiprianDraghici) [reply package.json:23]: we can't as we are using specific things from v13 (see `signMessage`)

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**